### PR TITLE
Add retirement_bulk_status_update script and support

### PR DESCRIPTION
**Pull request by bmedx**

### CODE REVIEW

File: tubular/scripts/retirement_bulk_status_update.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+# Return codes for various fail cases
+ERR_NO_CONFIG = -1
+ERR_BAD_CONFIG = -2
+ERR_FETCHING = -3
+ERR_UPDATING = -4
+ERR_SETUP_FAILED = -5
+
+
+def _config_or_exit(config_file):
```
Comment (doctoryes) [inline tubular/scripts/retirement_bulk_status_update.py]: Feels like both this `_config_or_exit` and `_setup_lms_or exit` should be placed in helpers.py at this point?

File: tubular/scripts/retirement_bulk_status_update.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        lms_base_url = config['base_urls']['lms']
+        client_id = config['client_id']
+        client_secret = config['client_secret']
+
+        config['lms_api'] = LmsApi(lms_base_url, lms_base_url, client_id, client_secret)
+    except Exception as exc:  # pylint: disable=broad-except
+        FAIL(ERR_SETUP_FAILED, text_type(exc))
+
+
+def validate_dates(_, __, value):
```
Comment (doctoryes) [inline tubular/scripts/retirement_bulk_status_update.py]: Curious enough to ask about the first two params - but too lazy to look it up myself. Bah - doesn't matter...

File: tubular/scripts/retirement_bulk_status_update.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    help='(YYYY-MM-DD) Earliest creation date for retirements to act on.'
+)
+@click.option(
+    '--end_date',
+    callback=validate_dates,
+    help='(YYYY-MM-DD) Latest creation date for retirements to act on.'
+)
+def update_statuses(config_file, initial_state, new_state, start_date, end_date):
+    """
+    Does the bulk update of statuses
```
Comment (doctoryes) [inline tubular/scripts/retirement_bulk_status_update.py]: `Bulk-updates user retirement statuses where in a specified state -and- retirement was requested between a start date and end date.`?

File: tubular/scripts/retirement_bulk_status_update.py (hunk 1 above)
Comment (bmedx) [reply tubular/scripts/retirement_bulk_status_update.py]: Touché! Though there are two variants of the setup- one just does the yml, the other does yml + Google API, but that's solvable.

File: tubular/scripts/retirement_bulk_status_update.py (hunk 2 above)
Comment (bmedx) [reply tubular/scripts/retirement_bulk_status_update.py]: They are the context and parameter name, fwiw. I always go back and forth whether to do this or explicitly ignore them in pylint.

**Review by doctoryes — APPROVED** [review]

**Comment by bmedx** [comment]
@pwnage101 @doctoryes I had to rebase this to get in the unicode changes and everything else that went in while I was away. Can you take another look to make sure I didn't miss anything?

**Review by doctoryes — APPROVED** [review]
I'm unable to tell what changed in the rebase. But I looked over the PR again and still LGTM.

**Pull request merged**